documentation/MIGRATION guide for bumping up to v30.0.0.md

```diff
@@ -0,0 +1,196 @@
+1. [@azure/arm-features](https://www.npmjs.com/package/@azure/arm-features/v/30.0.0-beta.2)
+1. [@azure/arm-locks](https://www.npmjs.com/package/@azure/arm-locks/v/30.0.0-beta.1)
+1. [@azure/arm-links](https://www.npmjs.com/package/@azure/arm-links/v/30.0.0-beta.1)
+1. [@azure/arm-policy](https://www.npmjs.com/package/@azure/arm-policy/v/30.0.0-beta.1)
+1. [@azure/arm-managedapplications](https://www.npmjs.com/package/@azure/arm-managedapplications/v/30.0.0-beta.1)
+
+## Authentication
+
+In version 30.0.0+ Javescript/Typescript packages, we only `@azure/identity` to do the Authentication. And we have deprecated the authentication method of `@azure/ms-rest-nodeauth` and `@azure/ms-rest-browserauth`. If you are using them, you may change it accordingly.  
+For example, if you are using `loginWithServicePrincipalSecret` method in `@azure/ms-rest-nodeauth` to get the credential, you may replace it with `ClientSecretCredential` in `@azure/identity`.  
+
+change
+```typescript
+import * as msRestNodeAuth  from '@azure/ms-rest-nodeauth';
+const credentials = msRestNodeAuth.loginWithServicePrincipalSecret(clientId, clientSecret, tenantId);
+```
+into
+```typescript
+import { ClientSecretCredential }  from '@azure/identity';
+const credentials = new ClientSecretCredential(tenantId, clientId, clientSecrat);
+```
+Refer to [@azure/identity](https://www.npmjs.com/package/@azure/identity) for more details.
+
+
+## Callbacks
+
+In version below 30.0.0. we have some operations that allow users to use callback like :
+
+```typeScript
+getInstanceView(
+  resourceGroupName: string,
+  cloudServiceName: string,
+  callback: msRest.ServiceCallback<Models.CloudServiceInstanceView>
+): void;
+
+getInstanceView(
+  resourceGroupName: string,
+  cloudServiceName: string,
+  options: msRest.RequestOptionsBase,
+  callback: msRest.ServiceCallback<Models.CloudServiceInstanceView>
+): void;
+
+getInstanceView(
+  resourceGroupName: string,
+  cloudServiceName: string,
+  options?: msRest.RequestOptionsBase | msRest.ServiceCallback<Models.CloudServiceInstanceView>,
+  callback?: msRest.ServiceCallback<Models.CloudServiceInstanceView>
+): Promise<Models.CloudServicesGetInstanceViewResponse> 
+```
+
+Now, we have removed them, if you are using something like 
+```typescript
+const callback = function handle(...args) {
+    // callback function body
+}
+getInstanceView(resourceGroupName, cloudServiceName, callback);
+```
+You may change it into a promise based 
+```typescript
+const callback = function handle(...args) {
+    // callback function body
+}
+getInstanceView(resourceGroupName, cloudServiceName)
+  .then(...args => handle(...args))
+```
+
+## List Operations
+
+List operations now return an iterable result that follows the `PagedAsyncIterableIterator` interface as opposed to the previous model where you had to make a new request using the link to the next page.  
+which means previously, you can get the list result directly like
```

Comment:
Can we use sth like:
The below example shows how you could iterate through the list of items in previous version: